wp-cache-sanitizer.php

```diff
@@ -0,0 +1,726 @@
+   */
+  private function insert( $setting, $sanitized_value ) {
+    global $wp_cache_config_file;
+
+    if ( in_array( $setting, $this->allowed_booleans ) ) {
+        wp_cache_replace_line('^ *\$' . $setting . ' =', "\$" . $setting . " = " . $sanitized_value . ";", $wp_cache_config_file);
+    } else if ( in_array( $setting, $this->allowed_strings ) ) {
+        if ( is_array( $sanitized_value ) && $setting === 'wp_cache_pages' ) {
+            foreach ( $sanitized_value as $page => $status ) {
+                wp_cache_replace_line('^ *\$wp_cache_pages\[ "' . $page . '" \]', "\$wp_cache_pages[ \"{$page}\" ] = $status;", $wp_cache_config_file);
+            }
+        } else {
+            wp_cache_replace_line('^ *\$' . $setting . ' =', "\$". $setting ." = $sanitized_value;", $wp_cache_config_file);
+        }
+    } else if ( in_array( $setting, $this->allowed_integers ) ) {
+        wp_cache_replace_line('^ *\$' . $setting . ' =', "\$". $setting ." = $sanitized_value;", $wp_cache_config_file);
+    } else if ( in_array( $setting, $this->allowed_options ) ) {
+        update_option( $setting, $sanitized_value );
+    }
+  }
+
+  /**
+   * This sanitizes the settings where the allowed values are boolean, 0 or 1. The boolean values true and false
+   * are converted to strings to maintain the values in wp-cache-config.php file
+   *
+   * @param String $setting The current boolean setting to be sanitized
+   * @param Mixed $value The mixed value of the current setting that will be sanitized
+   *
+   * @return Boolean|String A sanitized version of the inputed value
+   */
+  function sanitize_boolean( $setting, $value ) {
+    // Note: necessary to force cache_enable and super_cache_enabled to be true/false
+    $sanitized_value = apply_filters( "wp_super_cache_sanitize_$value", $value );
+    if ( is_bool( $value ) === true ) {
+        $sanitized_value = $value ? 'true' : 'false';
+    } else {
+        $sanitized_value = intval( $value ) === 1 ? 1 : 0;
+    }
+    return $sanitized_value;
+  }
+
+  /**
+   * This sanitizes the settings where the allowed values are only integers.
+   * The sanitation method is different per setting name.
+   *
+   * @param String $setting The current integer setting to be sanitized
+   * @param Mixed $value The mixed value of the current setting that will be sanitized
+   *
+   * @return Integer A sanitized version of the inputed value
+   */
+  function sanitize_int( $setting, $value ) {
+    switch ( $setting ) {
+        case "cache_badbehaviour":
+            $value = $value == '' || $value == 'no' ? 0 : $value;
+            $sanitized_value = intval( $value );
+            break;
+
+        case "cache_awaitingmoderation":
+        case "cache_wptouch":
+        case "cache_domain_mapping":
+            $value = $value == '' ? '0' : $value;
+            $sanitized_value = intval( $value );
+            break;
+
+        case "cache_time_interval":
+            $sanitized_value = 600;
```

Comment:
This is hard to grok.  The following would be much easer for me to read.

`$sanitized_value = is_numeric( $value ) ? intval( $value ) : 600;`